src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/AdvSimd.cs

```diff
@@ -51,25 +51,25 @@ public abstract class AdvSimd : ArmBase
         //     /// float32x2_t vmla_lane_f32 (float32x2_t a, float32x2_t b, float32x2_t v, const int lane)
         //     ///   A32: VMLA.F32 Dd, Dn, Dm[lane]
         //     /// </summary>
-        //     public static Vector64<float> MultiplyAddBySelectedScalar(Vector64<float> addend, Vector64<float> left, Vector64<float> right, byte rightIndex) => MultiplyAddBySelectedScalar(addend, left, right, rightIndex);
+        //     public static Vector64<float> MultiplyAddBySelectedScalar(Vector64<float> addend, Vector64<float> left, Vector64<float> right, [ConstantExpected(Max = (byte)(1))] byte rightIndex) => MultiplyAddBySelectedScalar(addend, left, right, rightIndex);
```

Comment:
Arm64 and WASM expect values to be exactly in range. As such, we strictly limit the input between `0` and `Max` or `1` and `Max`.

The exact semantics is dependent on the instruction and is table driven in the JIT: https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/hwintrinsicarm64.cpp#L215-L271